test/storage_init_test.go

```diff
@@ -338,9 +340,212 @@ var _ = Describe(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						err = deleteCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should change the pvcUID in initialisedVolumes", func() {
+
+						aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						// validate initialisedVolumes format
+						err = validateInitVolumes(aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						initVolName := "ns"
+						unscheduledPod := clusterName + "-1-0"
+						deletedPVC := initVolName + "-" + unscheduledPod
+
+						oldInitVol := aeroCluster.Status.Pods[unscheduledPod].InitializedVolumes
+
+						By("Unscheduling aerospike pods")
+
+						aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = unschedulableResource()
+
+						err = updateClusterWithTO(k8sClient, ctx, aeroCluster, 1*time.Minute)
+						Expect(err).Should(HaveOccurred())
+
+						By("Cleaning up previous pvc")
+
+						pvcNamespacedName := getNamespacedName(
+							deletedPVC, namespace,
+						)
+
+						err = deletePVC(k8sClient, pvcNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Setting resources to schedule the pods")
+
+						aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = schedulableResource("200Mi")
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						newInitVol := aeroCluster.Status.Pods[unscheduledPod].InitializedVolumes
+
+						By("Validating initialisedVolumes list")
+
+						err = compareInitialisedVolumes(oldInitVol, newInitVol, initVolName)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should remove old formatted initialisedVolumes names", func() {
+
+						aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						// validate initialisedVolumes format
+						err = validateInitVolumes(aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						unscheduledPod := clusterName + "-1-0"
+
+						oldInitVol := aeroCluster.Status.Pods[unscheduledPod].InitializedVolumes
+
+						By("Patching old volume name format")
+
+						patch1 := jsonpatch.PatchOperation{
+							Operation: "replace",
+							Path:      "/status/pods/" + unscheduledPod + "/initializedVolumes",
+							Value:     append(oldInitVol, "ns"),
+						}
+
+						var patches []jsonpatch.PatchOperation
+						patches = append(patches, patch1)
+
+						jsonPatchJSON, err := json.Marshal(patches)
+						Expect(err).ToNot(HaveOccurred())
+
+						constantPatch := crClient.RawPatch(types.JSONPatchType, jsonPatchJSON)
```

Comment:
We can modify this test-case as per new flow like add old format pvcname and remove api-version label. Reconcile should add label and migrate volume names. Validate both initvolumes name and label